src/join.rs

```diff
@@ -113,6 +129,98 @@ impl<J: Join> std::iter::Iterator for JoinIter<J> {
     }
 }
 
+/// `JoinParIter` is a `ParallelIterator` over a group of `Storages`.
+#[must_use]
+pub struct JoinParIter<J>(J);
+
+impl<J> ParallelIterator for JoinParIter<J>
+  where
+    J: Join + Send,
+    J::Type: Send,
+    J::Value: Split + Send,
+    J::Mask: Send + Sync,
+{
+    type Item = J::Type;
+
+    fn drive_unindexed<C>(self, consumer: C) -> C::Result
+        where C: UnindexedConsumer<Self::Item>
+    {
+        let (keys, values) = self.0.open();
+        let producer = BitProducer((&keys).iter());
+        let values = UnsafeCell::new(values);
+        bridge_unindexed(JoinProducer::<J>::new(producer, &values), consumer)
+    }
+}
+
+struct JoinProducer<'a, J>
+  where
+    J: Join + Send,
```

Comment:
It would be nice to cut down bounds on the struct to a minimum. You could probably just do `JoinProducer<'a, M, V>` or something. The reason for this convention is that it would force you to add these bounds everywhere, infecting all the types using it.

Consider this struct:

```rust
struct BoxMap<K, V, H> {
    map: HashMap<K, V, H>,
}
```

So no bounds necessary in the struct declaration.